pkg/util/deployment/deployment.go

```diff
@@ -61,7 +61,11 @@ func GetOldRCs(deployment extensions.Deployment, c client.Interface) ([]*api.Rep
 		}
 	}
 	requiredRCs := []*api.ReplicationController{}
-	for _, value := range oldRCs {
+	// Note that go reuses the same memory location for every iteration,
+	// which means the 'value' returned from range will have the same address.
+	// Therefore, we should use the returned 'index' instead.
+	for i := range oldRCs {
```

Comment:
I would call this a key instead of an index since this is a map and not a slice.